vault/raft.go

```diff
@@ -1116,3 +1151,14 @@ type answerResp struct {
 	Peers      []raft.Peer      `json:"peers"`
 	TLSKeyring *raft.TLSKeyring `json:"tls_keyring"`
 }
+
+func newDiscover() (*discover.Discover, error) {
```

Comment:
Is this helper still needed now that we're not adding k8s?

Reply:
Yeah we can safely remove this 👍

Reply:
But on second thought, just in case the go-discover API changes, I figure it's safe to be explicit. So I'm leaning towards keeping this.